File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int curr=nums[0],next=0;
        for(int i=1;i<nums.size();i++)
        {
            if(i<=curr)
            {
                next=max(next,i+nums[i]);
            }
            else
            {
                curr=next;
                next=nums[i]+i;
                if(curr<i)
                    return false;
            }
        }
        return true;
    }
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int reach=nums[0];
        for(int i=1;i<nums.length;i++)
        {
            if(reach==0)
                return false;
            reach--;
            reach=Math.max(reach,nums[i]);
        }
        return true;
    }
}